cmd/auto-complete.go

```diff
@@ -26,6 +26,24 @@ import (
 	"github.com/posener/complete"
 )
 
+const escapeCharacters = "\\ !\"$&'()*;<=>?@[\\]^`{|}"
+
+// escaper predicts the completion using the underlying
+// Predictor function and escapes all found predictions.
+type escaper struct {
+	complete.Predictor
+}
+
+func (e escaper) Predict(args complete.Args) []string {
+	predictions := e.Predictor.Predict(args)
+	for i := range predictions {
+		for _, escape := range escapeCharacters {
+			predictions[i] = strings.Replace(predictions[i], string(escape), "\\"+string(escape), -1)
```

Comment:
What if parts are already escaped?

So if you have `file\"name<autocomplete>` - will it not double escape the `\"`?

Reply:
> So if you have `file\"name<autocomplete>` - will it not double escape the `\"`?

yes you are right, this is broken and does not work in all cases.

I think we have two options here:
- We need to send a PR to the original completion library (posener/complete) to accept some special config to make bash auto-escape filenames but the library supports multiple shells (zsh, fish, ..) at the same time and I am not sure if this is doable.

Another option is we do it and make our code more sophisticated.

I am trying the second option for now and see how easy it is